crates/vizia_core/src/context/mod.rs

```diff
@@ -236,6 +238,19 @@ impl Context {
         self.focus_with_visibility(old_focus_visible)
     }
 
+    /// Stop the user from tabbing out of a subtree, which is useful for modals.
+    pub fn lock_focus_to(&mut self, lock_focus_to: Entity) {
+        self.lock_focus_to = lock_focus_to;
+        // Adjust the currently focused entity if it is not currently within the subtree
+        if !self.focused.is_descendant_of(&self.tree, lock_focus_to) {
+            let new_focus = TreeIterator::subtree(&self.tree, lock_focus_to)
+                .filter(|node| is_focusable(&self, *node))
+                .next()
+                .unwrap_or(Entity::root());
+            self.with_current(new_focus, |cx| cx.focus());
+        }
```

Comment:
If a developer has a particular entity they want to have focused and which they call focus on before the lock focus is called than I wanted to retain that choice.

Reply:
Okay I think I understand. So this would be useful if say the user had a textbox inside a modal and they wanted the textbox to gain focus when the modal is shown? They would set focus on the textbox and then lock the focus on the modal?